mesheryctl/pkg/utils/helpers_test.go

```diff
@@ -0,0 +1,506 @@
+	}
+}
+
+// func TestNavigateToBrowser(t *testing.T) {
+// 	// opens up a browser window whenever this test runs
+// 	err := NavigateToBrowser("https://www.layer5.io")
+// 	if err != nil {
+// 		t.Errorf("NavigateToBrowser error: %v", err)
+// 	}
+// }
+
+func TestUploadFileWithParams(t *testing.T) {
+	fixtureFileName := "listmanifest.api.response.golden" // any arbitrary fixture file
+	uploadFilePath := filepath.Join(fixturesDir, fixtureFileName)
+
+	test := struct {
+		uri       string
+		params    map[string]string
+		paramName string
+		path      string
+	}{
+		uri:       "https://www.layer5.io",
+		params:    nil,
+		paramName: "meshery",
+		path:      uploadFilePath,
+	}
+
+	// returns *http.Request
+	_, err := UploadFileWithParams(test.uri, test.params, test.paramName, test.path)
+
+	if err != nil {
+		t.Errorf("UploadFileWithParams error = %v", err)
+	}
+}
+
+func TestContentTypeIsHTML(t *testing.T) {
+	tests := []struct {
+		name           string
+		response       *http.Response
+		expectedOutput bool
+	}{
+		{
+			name: "correct content-type",
+			response: &http.Response{
+				Header: http.Header{
+					"Content-Type": []string{"text/html"},
+				},
+			},
+			expectedOutput: true,
+		},
+		{
+			name: "empty content-type",
+			response: &http.Response{
+				Header: http.Header{
+					"Content-Type": []string{},
+				},
+			},
+			expectedOutput: false,
+		},
+		{
+			name: "incorrect content-type",
+			response: &http.Response{
+				Header: http.Header{
+					"Content-Type": []string{"multipart/form-data"},
+				},
+			},
+			expectedOutput: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := ContentTypeIsHTML(tt.response)
+			if got != tt.expectedOutput {
+				t.Errorf("ContentTypeIsHTML error = %v want = %v", got, tt.expectedOutput)
+			}
+		})
+	}
+}
+
+func TestAskForConfirmation(t *testing.T) {
+	tests := []struct {
+		name     string
+		question string
+		input    string
+		want     bool
+	}{
+		{
+			name:     "test with input 'yes'",
+			question: "question?",
+			input:    "yes\n",
+			want:     true,
+		},
+		{
+			name:     "test with input 'no'",
+			question: "question?",
+			input:    "no\n",
+			want:     false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// mocking stdio
+			// see https://stackoverflow.com/a/64518829
+			r, w, err := os.Pipe()
+			if err != nil {
+				t.Fatal(err)
+			}
+			_, err = w.WriteString(tt.input)
+			if err != nil {
+				t.Error(err)
+			}
+			w.Close()
+			stdin := os.Stdin
+			defer func() { os.Stdin = stdin }()
+			os.Stdin = r
+
+			got := AskForConfirmation(tt.question)
+			if got != tt.want {
+				t.Errorf("AskForConfirmation got = %v want = %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestCreateConfigFile(t *testing.T) {
+	originalDefaultConfigPath := DefaultConfigPath
+	defer func() { DefaultConfigPath = originalDefaultConfigPath }()
+	DefaultConfigPath = "config.yaml"
```

Comment:
setup path like this. os-home-path + ".meshery" + "config.yaml"

Reply:
Right. The thing is, this path is there by default. We don't have to explicitly set it. But I think this failed because in our test environment, `$HOME/.meshery/` directory doesn't exist. That's why I commented this test in the first place.

Reply:
Done!